File: AndroidStudioProjects/OOPProject/app/src/main/java/com/ecrinaybay/oopproject/MainActivity.java
package com.ecrinaybay.oopproject;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        /*User myUser = new User();
        myUser.name = "Atil";
        myUser.job = "Instrucor";

        User newUser = new User();
        newUser.name = "Kirk";
        newUser.job = "Musician";*/

        User myUser = new User("Atil","Instructor");
        System.out.println(myUser.information());
        System.out.println(myUser.name);

        //Encapsulations

        Musicians james = new Musicians("James","Guitar",50);
        System.out.println(james.getName());
        james.setAge(60,"Kirk"); //age'i değiştiremeyiz çünkü password yanlış.
        james.setAge(60,"Atil"); //password doğru olduğu için age değiştirilir.
        System.out.println(james.getAge());

        //Inheritance

        SuperMusicians lars = new SuperMusicians("Lars","Drums",55);
        System.out.println(lars.sing());
        System.out.println(lars.getAge());

        //Polymorphism

        //Static Polymorphism

        Mathematics mathematics = new Mathematics();
        System.out.println(mathematics.sum());
        System.out.println(mathematics.sum(5,3));
        System.out.println(mathematics.sum(5,3,4));

        //Dynamic Polymorphism

        Animal myAnimal = new Animal();
        myAnimal.sing();
        Dog barley = new Dog();
        barley.test();
        barley.sing();

        Piano myPiano = new Piano();
        myPiano.brand = "Yamaha";
        myPiano.digital = true;
        myPiano.info();


    }
}

File: AndroidStudioProjects/CountDownKotlin/app/src/main/java/com/ecrinaybay/countdownkotlin/MainActivity.kt
package com.ecrinaybay.countdownkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import com.ecrinaybay.countdownkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        //Abstract - Interface - Inheritance

        object : CountDownTimer(10000,1000){
            override fun onTick(millisUntilFinished: Long) {
                binding.textView.text = "Left:${millisUntilFinished / 1000}"
            }

            override fun onFinish() {
                binding.textView.text = "Left: 0"
            }

        }.start()

    }
}






























